src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,103 @@
+package duke.main;
+
+import duke.command.*;
+import duke.exception.DukeException;
+
+public class Parser {
+    public Parser() {
+    }
+
+    public Command parse(String[] fullCommand) throws DukeException {
+        //Add Find Command
+        String command = fullCommand[0];
+        String details = fullCommand[1];
+        if (command.equals("bye")) {
+            return parseBye(details);
+        } else if (command.equals("list")) {
+            return parseList(details);
+        } else if (command.equals("done")) {
+            return parseDone(details);
+        } else if (command.equals("delete")) {
+            return parseDelete(details);
+        } else if (command.equals("deadline")) {
+            return parseDeadline(details);
+        } else if (command.equals("event")) {
+            return parseEvent(details);
+        } else if (command.equals("todo")) {
+            return parseTodo(details);
+        } else {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-()");
+        }
+    }
+
+    public ByeCommand parseBye(String details) throws DukeException {
+        if (!details.isBlank()) {
+            throw new DukeException("OOPS!!! No input is needed after bye'");
+        }
+        return new ByeCommand();
+    }
+
+    public ListCommand parseList(String details) throws DukeException {
+        if (!details.isBlank()) {
+            throw new DukeException("OOPS!!! No input is needed after 'list'");
+        }
+        return new ListCommand();
+    }
+
+    public DoneCommand parseDone(String details) throws DukeException {
+        try {
+            int index = Integer.parseInt(details) - 1;
+            return new DoneCommand(index);
+        }
+        catch (Exception e) {
+            throw new DukeException("OOPS!!! Please input a valid number after 'done'");
+        }
+    }
+
+    public DeleteCommand parseDelete(String details) throws DukeException {
+        try {
+            int index = Integer.parseInt(details) - 1;
+            return new DeleteCommand(index);
+        }
+        catch (Exception e) {
```

Comment:
Catch `NumberFormatException` instead to be more specific as to what exception we are dealing with